Client/src/components/Layout/Desktop/index.js

```diff
@@ -1,23 +1,38 @@
 import React, { Component } from "react";
 import { BrowserRouter, Route, Switch, Redirect } from "react-router-dom";
-import Footer from "../DesktopFooter";
-import SideBar from "../SideBar";
-import Courses from "../../contanier/staff/Courses";
-import Student from "../../contanier/staff/Student";
-import auth from "./../../auth";
+import Footer from '../DesktopFooter';
+import SideBar from '../../common/Sidebar';
+import Student from '../../contanier/staff/Student';
+import Viewcourse from '../../contanier/staff/Courses/ViewCourse';
+import AddCourse from '../../contanier/staff/Courses/AddCourse'
+import ViewCourseDetails from '../../contanier/staff/Courses/ViewCourse-Details';
+import AddStudent from "../../contanier/staff/Student/AddStudent";
+import ViewStudent from "../../contanier/staff/Student/ViewStudent";
+import EditStudent from "../../contanier/staff/Student/EditStudent";
 
 export default class Desktop extends Component {
-  state = {};
+  state = {
+  }
   render() {
-    const { role } = auth.isAuthenticated();
-    if (role === "student") this.props.history.push("/student");
     return (
-      <>
-        <SideBar />
-        <Route exact path={"/staff/courses"} component={Courses} />
-        <Route exact path={"/staff/student"} component={Student} />
-        <Footer />
-      </>
+      <BrowserRouter className='desktop-container'>
+        <div className='contanier'>
+          <SideBar />
+          <div>
+            <Switch>
+              <Route exact path={'/staff/courses/'} render={(props) => <Viewcourse {...props} />} />
+              <Route exact path={'/staff/courses/viewcourse'} render={(props) => <Viewcourse {...props} />} />
+              <Route exact path={'/staff/courses/addcourse'} render={(props) => <AddCourse {...props} />} />
+              <Route exact path={'/staff/courses/:id'} render={(props) => <ViewCourseDetails  {...props} />} />
+              <Route exact path={'/staff/courses/viewcourse/:id'} render={(props) => <ViewCourseDetails  {...props} />} />
```

Comment:
@Fatmasiam These two routes (/staff/fourses/:id, /staff/courses/viewcourse/:id) look like they show the same component - so you could probably just have one instead of two.